components/org.wso2.carbon.config/src/main/java/org/wso2/carbon/config/ConfigProviderFactory.java

```diff
@@ -0,0 +1,113 @@
+
+    /**
+     * Initializes and returns configuration provider service from the configuration file provided.
+     *
+     * @param filePath configuration absolute filepath(e.g: {carbon-home}/conf/deployment.yaml})
+     * @return configProvider service object.
+     * @throws ConfigurationException if an error occurred while initializing the config provider.
+     */
+    public static ConfigProvider getConfigProvider(Path filePath) throws ConfigurationException {
+        return getConfigProvider(filePath, getSecureVaultService(filePath));
+    }
+
+    /**
+     * Initializes and returns configuration provider service with the provided configuration file.
+     *
+     * @param filePath configuration absolute filepath(e.g: {carbon-home}/conf/deployment.yaml})
+     * @param secureVault {@code SecureVault>}
+     * @return configProvider service object
+     * @throws ConfigurationException if filepath == null or securevault == null or configuration file extension is
+     * not equal to xml or yaml.
+     */
+    public static ConfigProvider getConfigProvider(Path filePath, SecureVault secureVault) throws
+            ConfigurationException {
+        //check whether configuration filepath is null. proceed if not null.
+        if (filePath == null || !filePath.toFile().exists()) {
+            throw new ConfigurationException("No configuration filepath is provided. configuration provider will " +
+                    "not be initialized!");
+        }
+        //check whether securevault is null. proceed if not null.
+        if (secureVault == null) {
+            throw new ConfigurationException("No securevault service found. configuration provider will not be " +
+                    "initialized!");
+        }
+        if (logger.isDebugEnabled()) {
+            logger.debug("initialize config provider instance from configuration file: " + filePath.toString());
+        }
+        // initialize config provider service from the configuration file provided.
+        ConfigFileReader configFileReader;
+        if (filePath.toString().endsWith(YAML_EXTENSION)) {
+            configFileReader = new YAMLBasedConfigFileReader(filePath);
+        } else if (filePath.toString().endsWith(XML_EXTENSION)) {
+            configFileReader = new XMLBasedConfigFileReader(filePath);
+        } else {
+            throw new ConfigurationException("Error while initializing configuration provider, file extension is not " +
+                    "supported");
+        }
+        return new ConfigProviderImpl(configFileReader, secureVault);
+    }
+
+    /**
+     * Load {@code SecureVault} implementation in non-osgi mode.
+     *
+     * @param filePath configuration absolute filepath(e.g: {carbon-home}/conf/deployment.yaml})
+     * @throws ConfigurationException if an error occurred while loading securevault service.
+     */
+    private static SecureVault getSecureVaultService(Path filePath) throws ConfigurationException {
```

Comment:
why this method is named "getSecureVaultService"? 
It could be "getSecureVault()"